src/plugins/intel_cpu/src/nodes/fake_quantize.cpp

```diff
@@ -1888,153 +1880,205 @@ void FakeQuantize::appendPostOps(dnnl::post_ops& ops, const VectorDims &postOpDi
-        std::all_of(osh.cbegin(), osh.cend(), [](float val) { return std::abs(val + 128.f) < 0.0001f; })) {
-        bool isCropAligned = true;
-        for (int i = 0; i < std::max(cl.size(), isc.size()); i++) {
-            if (std::abs(cl[cl.size() == 1 ? 0 : i] * isc[isc.size() == 1 ? 0 : i] + 128.f) > 0.0001f) {
-                isCropAligned = false;
-            }
+    const size_t bufferAlignment = 16;
+    initializePostOpData(dnnlpoc.getOutputDims(), bufferAlignment, doRounding);
+
+    auto& f = optimizedFormula;
+
+    DEBUG_LOG("\t ---- Optimized formula ----");
+    DEBUG_LOG("\t inputScale =[", PrintableVector<float>(f.isc), "]");
+    DEBUG_LOG("\t inputShift =[", PrintableVector<float>(f.ish), "]");
+    DEBUG_LOG("\t    cropLow =[", PrintableVector<float>(f.clo), "]");
+    DEBUG_LOG("\t   cropHigh =[", PrintableVector<float>(f.chi), "]");
+    DEBUG_LOG("\toutputScale =[", PrintableVector<float>(f.osc), "]");
+    DEBUG_LOG("\toutputShift =[", PrintableVector<float>(f.osh), "]");
+
+    // when FQ is last postOps and output data type is u8/s8
+    // round & clip2 can be further optimized since saturation will be performed by oneDNN by default
+    bool skipRoundClipOutputLinear = false;
+    if (isLastPostOp && (levels == 256) && f.clo.size() == 1 && f.chi.size() == 1 && f.osc.empty() && f.osh.empty()) {
+        if (outDataType == memory::data_type::u8 && f.clo[0] <= 0.0f && f.chi[0] >= 255.0f) {
+            skipRoundClipOutputLinear = true;
         }
-
-        for (int i = 0; i < std::max(ch.size(), isc.size()); i++) {
-            if (std::abs(ch[ch.size() == 1 ? 0 : i] * isc[isc.size() == 1 ? 0 : i] - 127.f) > 0.0001f) {
-                isCropAligned = false;
-            }
+        if (outDataType == memory::data_type::s8 && f.clo[0] <= -128.0f && f.chi[0] >= 127.0f) {
+            skipRoundClipOutputLinear = true;
         }
+    }
 
-        if (isCropAligned) {
-            outScale = isc;
-            if (!outScale.empty()) {
-                size_t size = outScale.size();
-                if (size == 1 && Shape::UNDEFINED_DIM != OC) {
-                    outScale.resize(OC);
-                    for (size_t k = 0; k < OC; k++)
-                        outScale[k] = outScale[0];
-                }
-            }
+    // return false before committing any change to DnnlPostOpsComposer
+    if (!allowBinary) {
+        if (f.ish.size() > 1)
+            return false;
+        if (!skipRoundClipOutputLinear) {
+            if (f.clo.size() > 1 || f.chi.size() > 1)
+                return false;
+            if (f.osc.size() > 1 || f.osh.size() > 1)
+                return false;
         }
     }
 
-    return outScale;
+    if (!dnnlpoc.appendLinear(f.isc, f.ish, allowBinary))
+        return false;
+
+    if (skipRoundClipOutputLinear)
+        return true;
+
+    if (doRounding)
+        dnnlpoc.appendRoundHTE();
+    dnnlpoc.appendClip(f.clo, f.chi);
+    dnnlpoc.appendLinear(f.osc, f.osh, allowBinary);
```

Comment:
As I explained in another comment below, the check done here would be too late, since we cannot undo the input-scale/shift, round and clip that have been appended into post-ops/attr, so I choose to avoid failure in advance, at the very beginning, before any change was made to post-ops  (in the `if (!allowBinary) {` block) , so if we can reach here, this call should always succeed.